src/search/lp/lp_solver.h

```diff
@@ -73,6 +76,28 @@ struct LPVariable {
                double objective_coefficient);
 };
 
+class LinearProgram {
+    LPObjectiveSense sense;
+    std::string objective_name;
+
+    utils::NamedVector<LPVariable> variables;
+    utils::NamedVector<LPConstraint> constraints;
+
+public:
+    explicit LinearProgram(LPObjectiveSense sense, utils::NamedVector<LPVariable> &&variables, utils::NamedVector<LPConstraint> &&constraints, std::string objective_name = "")
+        : sense(sense), variables(std::move(variables)), constraints(std::move(constraints)), objective_name(objective_name) {
+    }
+
+    utils::NamedVector<LPVariable> &get_variables();
```

Comment:
Please add some documentation explaining that variables and constraints can be named but doing so will have an impact on performance, so this should only be done for debugging purposes.